### DISCUSSION TITLE
Creating Developer Materials - Video Series

### DISCUSSION THREAD

**Discussion by nblogist** [ringdao-community-guild/discussion#167] [General]
# Proposal: Darwinia Network Developer Education Series

## Introduction

As Darwinia Network continues to grow and evolve, one of our primary objectives is to significantly increase the number of developers deploying applications on our platform. This goal is a key component of our ongoing marketing initiatives and overall ecosystem expansion strategy.

To ensure the success of this endeavor, we recognize the critical importance of providing a seamless onboarding experience for new developers. While we currently have documentation available, we believe that supplementing this with a comprehensive, video-based educational series will greatly enhance the learning experience and reduce the potential for high bounce rates as we attract more developers to our ecosystem.

The proposed video series aims to be:

1. Concise yet comprehensive
2. Demonstrative and practical
3. Easy to understand and follow
4. Complementary to our existing documentation

By creating this resource, we aim to:

- Accelerate the onboarding process for new developers
- Reduce barriers to entry for building on Darwinia
- Showcase the unique features and capabilities of our network
- Foster a more engaged and knowledgeable developer community

This proposal outlines a tentative syllabus for this educational series. It is a work in progress and open to feedback from our community and team members. Bear and I will be taking the lead in producing this content, with the goal of creating a polished, professional series that requires minimal post-production editing. Special thanks to @PartnershipDude for pitching this.

We invite all stakeholders to review this proposal and provide their insights to help us create the most effective and valuable resource for our growing developer community.

## Introduction
This syllabus is designed for developers who already have basic knowledge of Solidity and blockchain development. It focuses on Darwinia-specific features, advanced concepts, and cross-chain functionalities.

### 1. Darwinia Network Overview
- Unique features and architecture
- How Darwinia differs from other networks

### 2. Darwinia Development Environment
- Darwinia-specific tools and frameworks
- Connecting to Darwinia testnet and mainnet

### 3. Smart Contract Deployment on Darwinia
- Differences in deployment process compared to Ethereum
- Darwinia-specific contract considerations

### 4. Cross-chain Functionality
- Implementing cross-chain token transfers
- Building cross-chain DApps

### 5. Substrate Integration
- How Solidity contracts interact with Substrate modules
- Leveraging Substrate features in Solidity contracts

### 6. Darwinia's Consensus Mechanism
- Understanding and interacting with Darwinia's consensus
- Building applications that leverage the consensus mechanism

### 7. Scalability Solutions **_(THIS ONE I AM NOT TOO SURE ABOUT YET)_**
- Layer 2 solutions specific to Darwinia
- Optimizing contracts for Darwinia's scalability features

### 8. Governance and Tokenomics
- Interacting with Darwinia's governance system
- Building applications that leverage RING and KTON tokens

### 9. Security Considerations
- Best practices for secure contract development on Darwinia
- Audit tools and processes specific to Darwinia

### 10. Advanced Darwinia Features
- Unique functionalities specific to Darwinia

#### 11.1 Introduction to XCMP
  - XCMP and its role in the Polkadot ecosystem
  - XCMP's relation to Darwinia's cross-chain functionality
  
  #### 11.2 XCMP Mechanics
  - Types of XCMP: Full XCMP, HRMP (XCMP Lite), and VMP
  - Message formats and protocols
  
  #### 11.3 Implementing XCMP in Darwinia
  - Sending and receiving cross-chain messages
  - Integrating XCMP into smart contracts
  
  #### 11.4 XCMP Use Cases
  - Walk through of Darwinia Ecosystem apps that use XCMP (Helix, MsgPort, etc.)
  - Cross-chain token transfers
  - Cross-chain smart contract calls
  - Building multi-chain DApps
  
  #### 11.5 XCMP Security and Best Practices
  - Understanding and mitigating XCMP-specific risks
  - Performance considerations and optimizations

## Course Delivery

Each topic will be covered through:
1. A short introductory video explaining the concept
2. A hands-on tutorial video showing implementation
3. Code samples and templates specific to Darwinia
4. A challenge or project idea for developers to practice
5. A list of common issues and their solutions

## Feedback and Iteration
I encourage developers to provide feedback on each module to help us continuously improve the course content and address the most relevant challenges faced by the Darwinia developer community. Please tag any active developers in the ecosystem that might be able to provide their input in this.

**Comment by nblogist** [comment]
Bear and I have a call regarding this and will post any updates here.

**Reply by boundless-forest** [reply]
1. Darwinia Network Overview

    - Unique features and architecture
    - How Darwinia differs from other networks
    - Darwinia chains(darwinia, crab, koi) introduction

2. Token Transfer on Darwinia

    - Introduce tokens on Darwinia chains
    - Connecting to Darwinia testnet and mainnet
    - Token transfer examples

3. Smart Contract Deployment on Darwinia
    
    - Differences in deployment process compared to Ethereum
    - Darwinia-specific contract considerations

4. Substrate Integration

    - How Solidity contracts interact with Substrate modules
    - Leveraging Substrate features in Solidity contracts

5. Cross-chain Functionality - Msgport

    - Implementing cross-chain token transfers
    - Building cross-chain DApps

6. Cross-chain Functionality - XCM

    - XCMP and its role in the Polkadot ecosystem
    - Message formats
    - Sending and receiving cross-chain messages


7. Darwinia's Staking Mechanism
    
    - Introduce Darwinia staking mechanism
    - How the users can stake on Darwinia

8. Governance and Tokenomics

    - Interacting with Darwinia's governance system
    - Building applications that leverage RING and KTON tokens

9. Security Considerations

    - Best practices for secure contract development on Darwinia
    - Audit tools and processes specific to Darwinia

10. Advanced Darwinia Features

    - Unique functionalities specific to Darwinia

**Reply by boundless-forest** [reply]
I reviewed the syllabus and made some adjustments to it. This is my revised version for your reference. @nblogist @PartnershipDude

**Reply by nblogist** [reply]
Let me review later today!

**Reply by nblogist** [reply]
Thank you! I see it is compressed. We will just have to expand more on the demonstrative parts but i see it as a good start to making scripts.

**Comment by nblogist** [comment]
@hackfisher please lmk if you have any feedback!

**Reply by SasoLithops** [reply]
Huge support of idea. Great one @PartnershipDude  and well done to @nblogist @boundless-forest to push this fwd. These video tutorials are so much better then written ones.
Will hackathlon follow to onboard some developers or how do we plan to make this videos noticed?

**Reply by boundless-forest** [reply]
Darinia will sponsor the Polkadot Hackathon 2024 Bounty https://docs.google.com/document/d/1cxsTVbUKIBpZsfYvG_0rMU-JxMec97rfuApiLXtfigA/edit#heading=h.lnhyt4gt4mya. If we can complete the series before then, we can showcase it at the Hackathon in Bangkok. I believe that would be a perfect place to present it.

**Reply by SasoLithops** [reply]
Perfect start 😉👍

**Reply by nblogist** [reply]
We also have plans to reach out to CS student bodies of universities around us.

**Reply by SasoLithops** [reply]
Awesome👏👏

**Comment by boundless-forest** [comment]
## Darwinia Network Overview

### Architecture

https://excalidraw.com/#json=EA8wPmSRZ-4l11fEX7NDW,3Cz2rCf0JZ-Qcl31UESXBA
### Features
- Cross-Chain Capabilities: https://docs.darwinia.network/#cross-chain-capabilities
- EVM Compatibility: https://docs.darwinia.network/#evm-compatibility
- Security and  Decentralized: https://docs.darwinia.network/#security-and-decentralized

### Chains

- https://docs.darwinia.network/build/getting-started/networks/overview/

**Reply by nblogist** [reply]
>Architecture
>https://excalidraw.com/#json=EA8wPmSRZ-4l11fEX7NDW,3Cz2rCf0JZ-Qcl31UESXBA

For developers coming from a solidity background we need an architecture diagram that either explains it in more details or juxtapose it with what a solidity developer is assumed to know.

**Reply by boundless-forest** [reply]
An updated version: https://excalidraw.com/#json=-lrLW2koDPl9voD5SxiYT,CazMuPABZkt8CaMuhrndKg

**Comment by nblogist** [comment]
> * Darwinia Network Overview
>   
>   * Unique features and architecture
>   * How Darwinia differs from other networks
>   * Darwinia chains(darwinia, crab, koi) introduction
> * Token Transfer on Darwinia
>   
>   * Introduce tokens on Darwinia chains
>   * Connecting to Darwinia testnet and mainnet
>   * Token transfer examples
> * Smart Contract Deployment on Darwinia
>   
>   * Differences in deployment process compared to Ethereum
>   * Darwinia-specific contract considerations
> * Substrate Integration
>   
>   * How Solidity contracts interact with Substrate modules
>   * Leveraging Substrate features in Solidity contracts
> * Cross-chain Functionality - Msgport
>   
>   * Implementing cross-chain token transfers
>   * Building cross-chain DApps
> * Cross-chain Functionality - XCM
>   
>   * XCMP and its role in the Polkadot ecosystem
>   * Message formats
>   * Sending and receiving cross-chain messages
> * Darwinia's Staking Mechanism
>   
>   * Introduce Darwinia staking mechanism
>   * How the users can stake on Darwinia
> * Governance and Tokenomics
>   
>   * Interacting with Darwinia's governance system
>   * Building applications that leverage RING and KTON tokens
> * Security Considerations
>   
>   * Best practices for secure contract development on Darwinia
>   * Audit tools and processes specific to Darwinia
> * Advanced Darwinia Features
>   
>   * Unique functionalities specific to Darwinia

Thanks to @boundless-forest for helping with these!
This is the one we are going on, and will be building the video series here!
